server/src/services/secret-codes.js

```
@@ -8,6 +10,39 @@ const codeGenerator = async (data, secret) => {
   return code;
 }
 
+const readKey = async () => {
+  // TODO: Store encrypted
+  const key = {
+    privateKey: process.env.OWNER_PRIVATE_KEY,
+    address: process.env.OWNER_ADDRESS,
+  }
+  return  key;
+}
+
+const signOffer = async (booking, key) => {
+  try {
+    const randomCode = Math.floor((1 + Math.random()) * 10000);
+    const bookingHash = web3.utils.sha3(`${randomCode}${Date.now()}`);
+    const hashedMessage = web3.utils.soliditySha3(
+      { type: 'string', value: booking.roomType||'double' },
+      { type: 'uint256', value: booking.paymentAmount },
+      { type: 'uint256', value: booking.signatureTimestamp },
+      { type: 'string', value: booking.paymentType },
+      { type: 'bytes32', value: bookingHash }
+    );
+
+    web3.eth.accounts.wallet.add(key);
+    const accounts = web3.eth.accounts.wallet;
+    const offerSignature = await web3.eth.sign(hashedMessage, accounts[0].address);
+    web3.eth.accounts.wallet.clear();
+    return offerSignature;
+  } catch (e) {
+    console.log(e);
```

you should propagate this error